Update avatar size prop

oluwatimio opened the pull request:
### Background

Updates avatar size prop to SizeKeyword. We do not have to specify small-100 and large-100 as these are aliases of small and large. We are following the same approach as admin here.

Also see https://github.com/Shopify/shopify-dev/pull/61707

### Solution

Updated typings and used SizeKeyword

### 🎩

<img width="2555" height="1257" alt="Screenshot 2025-08-18 at 3 13 54 PM" src="https://github.com/user-attachments/assets/d2c2e27c-fecd-4750-bc5e-7f62f6c060d1" />



### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

oluwatimio committed:
Update avatar size prop

oluwatimio committed:
Add changeset

oluwatimio committed:
Update guidance

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar/Avatar.doc.ts:
```diff
@@ -38,11 +38,12 @@ const data: ReferenceEntityTemplateSchema = {
-    * Large (39×39 px): Use when the avatar is a focal point, such as a customer details card.
-    * Extra-large (47x47 px): Use when placing more emphasis on the avatar
-    * Fill to fit: Use when there is a particular size that does not match any of the three sizes provided. If using images please ensure the resolution meets the size requirements.
+  - There are 5 sizes for the avatar component:
```
I will confirm this with @deannatron and @joao-quintino

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components.d.ts:
```diff
@@ -109,21 +109,24 @@ export interface AvatarProps extends IdProps {
    * @see https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload
    */
-  onLoad?(): void;
+  onLoad?(event: Event): void;
```
Same as what they do in checkout surface

andrewmcgov commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar.d.ts:
```diff
@@ -18,21 +18,24 @@ export interface AvatarProps extends IdProps {
+  size?: Extract<
+    SizeKeyword,
+    'small-200' | 'small' | 'base' | 'large' | 'large-200'
+  >;
```
Would just setting the type to `'small-200' | 'small' | 'base' | 'large' | 'large-200'` fix the order issue after it is compiled? Or does it need to be `SizeKeyword`?  I noticed that on your screenshot it doesnt actually say `SizeKeyword` there

andrewmcgov replied:
Yeah on the [2025-07 docs](https://shopify.dev/docs/api/customer-account-ui-extensions/latest/components/avatar) for Avatar docs it shows the type name: 

<img width="606" height="146" alt="Screenshot 2025-08-18 at 3 54 47 PM" src="https://github.com/user-attachments/assets/4624ac85-8a7b-4f83-901a-d84621a26bb7" />

I think we should be able to do that?

andrewmcgov replied:
I noticed that all of the checkout components have the same thing in the RC, the names of the types do not appear. I think we can disregard this for now.

oluwatimio committed:
Clean up doc

oluwatimio commented:
/snapit

oluwatimio commented:
I noticed the docs in Admin have an events section so I updated the docs here to also have that plus also cleaned up duplicate types. I will test these changes in an extension before shipping cc @robin-drexler @andrewmcgov 


<img width="2555" height="1257" alt="Screenshot 2025-08-18 at 5 53 41 PM" src="https://github.com/user-attachments/assets/8250785e-e138-4656-b2c4-24a55ca92c18" />

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar.d.ts:
```diff
@@ -1,6 +1,15 @@
+    })
+  | null;
+
+export interface AvatarElementProps extends IdProps {
```
These shouldnt have onLoad and onError since those are events

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar.d.ts:
```diff
@@ -13,35 +22,59 @@ export interface AvatarProps extends IdProps {
+  alt?: string;
+}
+
+export interface AvatarElementEvents {
```
Seperate events interface

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar.d.ts:
```diff
@@ -13,35 +22,59 @@ export interface AvatarProps extends IdProps {
+  onError?(event: Event): void;
+}
 
+export interface AvatarElement
```
The AvatarElement is a combo of AvatarElementProps and the AvatarElementEvents

oluwatimio commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar.d.ts:
```diff
@@ -13,35 +22,59 @@ export interface AvatarProps extends IdProps {
+  onerror: AvatarElementEvents['onError'];
+}
+
+export interface AvatarEvents {
```
The events are actually `load` and `error` so this is defined as well. Specifically used in the docs

oluwatimio commented:
Can confirm these type changes are working perfectly!! 

<img width="1560" height="318" alt="Screenshot 2025-08-19 at 10 03 33 AM" src="https://github.com/user-attachments/assets/b442ae7f-d106-4086-bf59-d8a75230884b" />
